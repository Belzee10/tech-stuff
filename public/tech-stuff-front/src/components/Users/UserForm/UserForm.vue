<template>
  <v-form ref="form" v-model="valid">
    <v-container>
      <v-row>
        <v-col cols="12">
          <v-text-field
            v-model="value.name"
            class="name"
            label="Name"
          ></v-text-field>
        </v-col>
        <v-col cols="12">
          <v-text-field
            v-model="value.lastName"
            class="last-name"
            label="Last Name"
          ></v-text-field>
        </v-col>
        <v-col cols="12">
          <v-text-field
            v-model="value.email"
            class="email"
            label="Email"
            type="email"
          ></v-text-field>
        </v-col>
        <v-col cols="12">
          <v-text-field
            v-model="value.password"
            class="password"
            label="Password"
            type="password"
          ></v-text-field>
        </v-col>
        <v-col cols="12">
          <v-select
            v-model="value.role"
            class="role"
            :items="['member', 'admin']"
            label="Role"
          ></v-select>
        </v-col>
        <v-col cols="12" class="d-flex justify-end">
          <v-btn
            :disabled="!valid"
            small
            class="submit text-capitalize primary"
            @click="submit"
            >Submit</v-btn
          >
          <v-btn small class="cancel text-capitalize ml-2" @click="cancel"
            >Cancel</v-btn
          >
        </v-col>
      </v-row>
    </v-container>
  </v-form>
</template>

<script>
export default {
  name: 'UserForm',
  props: {
    isEditable: {
      type: Boolean,
      default: false
    },
    populateWith: {
      type: Object,
      default: () => null
    }
  },
  data: () => ({
    valid: true,
    value: {
      name: '',
      lastName: '',
      email: '',
      password: '',
      role: ''
    }
  }),
  methods: {
    submit() {
      if (this.$refs.form.validate()) this.$emit('submit', this.value);
    },
    cancel() {
      this.$emit('cancel');
    }
  }
};
</script>

<style lang="scss" scoped></style>
