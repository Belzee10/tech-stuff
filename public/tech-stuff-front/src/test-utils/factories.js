import { mount, createLocalVue } from '@vue/test-utils';
import mergeWith from 'lodash.mergewith';
import Vuetify from 'vuetify';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);

let vuetify;

beforeEach(() => {
  vuetify = new Vuetify();
});

const customizer = (_, srcValue) => {
  if (Array.isArray(srcValue)) {
    return srcValue;
  }
  if (srcValue instanceof Object && Object.keys(srcValue).length === 0) {
    return srcValue;
  }
};

/**
 * create component wrapper
 * @param {*} Component
 * @param {Object} overrides
 */
const createWrapper = (Component, overrides) => {
  const defaultMountingOptions = {
    localVue,
    vuetify
  };
  return mount(
    Component,
    mergeWith(defaultMountingOptions, overrides, customizer)
  );
};

/**
 * create Vuex store
 * @param {Object} overrides
 */
const createStore = overrides => {
  const defaultStoreConfig = {};
  return new Vuex.Store(mergeWith(defaultStoreConfig, overrides, customizer));
};

export { createWrapper, createStore };
