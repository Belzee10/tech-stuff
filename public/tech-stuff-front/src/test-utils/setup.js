import Vue from 'vue';
import Vuetify from 'vuetify';

Vue.config.productionTip = false;

Vue.use(Vuetify);

/**
 * silent "console.error"
 */
global.console = {
  error: jest.fn(),
  log: console.log,
  warn: console.warn,
  info: console.info,
  debug: console.debug
};
