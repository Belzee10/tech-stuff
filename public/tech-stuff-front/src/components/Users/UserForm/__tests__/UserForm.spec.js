import { createWrapper } from '@/test-utils/factories';
import UserForm from '../UserForm.vue';

describe('UserForm.vue', () => {
  test('should render correctly', () => {
    const wrapper = createWrapper(UserForm);
    expect(wrapper.element).toMatchSnapshot();
  });
});
