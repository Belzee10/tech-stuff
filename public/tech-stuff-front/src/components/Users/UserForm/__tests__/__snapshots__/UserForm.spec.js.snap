// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm.vue should render correctly 1`] = `
<form
  class="v-form"
  novalidate="novalidate"
>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col col-12"
      >
        <div
          class="v-input name theme--light v-text-field"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <label
                  class="v-label theme--light"
                  for="input-4"
                  style="left: 0px; position: absolute;"
                >
                  Name
                </label>
                <input
                  id="input-4"
                  type="text"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <span
                  class="v-messages__wrapper"
                  name="message-transition"
                  tag="div"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="col col-12"
      >
        <div
          class="v-input last-name theme--light v-text-field"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <label
                  class="v-label theme--light"
                  for="input-7"
                  style="left: 0px; position: absolute;"
                >
                  Last Name
                </label>
                <input
                  id="input-7"
                  type="text"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <span
                  class="v-messages__wrapper"
                  name="message-transition"
                  tag="div"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="col col-12"
      >
        <div
          class="v-input email theme--light v-text-field"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <label
                  class="v-label theme--light"
                  for="input-10"
                  style="left: 0px; position: absolute;"
                >
                  Email
                </label>
                <input
                  id="input-10"
                  type="email"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <span
                  class="v-messages__wrapper"
                  name="message-transition"
                  tag="div"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="col col-12"
      >
        <div
          class="v-input password theme--light v-text-field"
        >
          <div
            class="v-input__control"
          >
            <div
              class="v-input__slot"
            >
              <div
                class="v-text-field__slot"
              >
                <label
                  class="v-label theme--light"
                  for="input-13"
                  style="left: 0px; position: absolute;"
                >
                  Password
                </label>
                <input
                  id="input-13"
                  type="password"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <span
                  class="v-messages__wrapper"
                  name="message-transition"
                  tag="div"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="col col-12"
      >
        <div
          class="v-input role theme--light v-text-field v-select"
        >
          <div
            class="v-input__control"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-owns="list-16"
              class="v-input__slot"
              role="button"
            >
              <div
                class="v-select__slot"
              >
                <label
                  class="v-label theme--light"
                  for="input-16"
                  style="left: 0px; position: absolute;"
                >
                  Role
                </label>
                <div
                  class="v-select__selections"
                >
                  <input
                    aria-readonly="true"
                    id="input-16"
                    readonly="readonly"
                    type="text"
                  />
                </div>
                <div
                  class="v-input__append-inner"
                >
                  <div
                    class="v-input__icon v-input__icon--append"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon notranslate mdi mdi-menu-down theme--light"
                    />
                  </div>
                </div>
              </div>
              <div
                class="v-menu"
              >
                <div
                  class="v-menu__content theme--light "
                  style="max-height: 304px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                />
              </div>
            </div>
            <div
              class="v-text-field__details"
            >
              <div
                class="v-messages theme--light"
              >
                <span
                  class="v-messages__wrapper"
                  name="message-transition"
                  tag="div"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="d-flex justify-end col col-12"
      >
        <button
          class="submit text-capitalize secondary v-btn v-btn--contained theme--light v-size--small"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            Submit
          </span>
        </button>
         
        <button
          class="cancel text-capitalize ml-2 v-btn v-btn--contained theme--light v-size--small"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            Cancel
          </span>
        </button>
      </div>
    </div>
  </div>
</form>
`;
